extensions/apps/extensions/src/components/pages/install-extension/install-extension-page.tsx

```diff
@@ -0,0 +1,225 @@
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import InstallApp, {
+  InstallAppProps,
+} from '@akashaorg/design-system-components/lib/components/InstallApp';
+import { useTranslation } from 'react-i18next';
+import { useAkashaStore, useRootComponentProps } from '@akashaorg/ui-awf-hooks';
+import ErrorLoader from '@akashaorg/design-system-core/lib/components/ErrorLoader';
+import Button from '@akashaorg/design-system-core/lib/components/Button';
+import { useGetAppsByPublisherDidQuery } from '@akashaorg/ui-awf-hooks/lib/generated';
+import getSDK from '@akashaorg/core-sdk';
+import { SortOrder } from '@akashaorg/typings/lib/sdk/graphql-types-new';
+import { GetAppsByPublisherDidQuery } from '@akashaorg/typings/lib/sdk/graphql-operation-types-new';
+import { getReportedError, getReportedProgress } from './utils';
+
+export const InstallExtensionPage = ({ appId }: { appId: string }) => {
+  const { t } = useTranslation('app-extensions');
+  const { getCorePlugins, decodeAppName, getRoutingPlugin } = useRootComponentProps();
+  const decodeName = React.useRef(decodeAppName);
+  const installer = useRef(getCorePlugins().extensionInstaller);
+  const installerStatusCodes = React.useRef(installer.current.getStaticStatusCodes());
+  const isInstalling = React.useRef(false);
+  const [isInstalled, setIsInstalled] = React.useState(false);
+  const [reportedStatus, setReportedStatus] = useState<symbol>();
+  const [reportedError, setReportedError] = useState<{ code: symbol; retryable: boolean }>(null);
+  const [shouldRegisterResources, setShouldRegisterResources] = useState<boolean>(false);
+  const idxDid = getSDK().services.gql.indexingDID;
+
+  const {
+    data: { authenticatedDID, isAuthenticating },
+  } = useAkashaStore();
+
+  const { data, loading, called, error } = useGetAppsByPublisherDidQuery({
+    variables: {
+      id: idxDid,
+      first: 1,
+      filters: { where: { name: { equalTo: decodeName.current(appId) } } },
+      sorting: { createdAt: SortOrder.Desc },
+    },
+  });
+
+  const retryableError: symbol[] = useMemo(
+    () => [
+      installerStatusCodes.current.error.EXTENSION_FETCH_ERROR,
+      installerStatusCodes.current.error.EXTENSION_IMPORT_ERROR,
+      installerStatusCodes.current.error.EXTENSION_INITIALIZATION_FAILED,
+      installerStatusCodes.current.error.EXTENSION_REGISTER_RESOURCES_FAILED,
+      installerStatusCodes.current.error.EXTENSION_INFO_SAVE_FAILED,
+      installerStatusCodes.current.error.EXTENSION_FINALIZATION_FAILED,
+    ],
+    [],
+  );
```

Comment:
i am curious cant we use `installerStatusCodes` directly instead of creating this ?